### ISSUE TITLE
Parser crashes when trying to parse POV demos

### ISSUE THREAD

**Issue by ivanhrabcak** [pnxenopoulos/csgo#74]
**Describe the bug**
I cannot parse any POV demos

**To Reproduce**
1. Record a POV demo with the `record <filename>` command in csgo
2. Try to parse it

[Generated JSON file](https://www.mediafire.com/file/xca0gqkn2d6y1ci/testdem-2_2.json/file)
[Example of a failing demo](https://www.mediafire.com/file/bio5ivwko167nlu/testdem-2_2.dem/file)  

**Expected behavior**
The demo should be parsed without an error.

**Log**
```
18:10:14 [INFO] Go version>=1.14.0
18:10:14 [INFO] Initialized CSGODemoParser with demofile /home/ivan/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/csgo/testdem-2_2.dem
18:10:14 [INFO] Setting demo id to testdem-2_2
18:10:14 [INFO] Setting parse rate to 128
18:10:14 [INFO] Setting trade time to 5
18:10:14 [INFO] Setting buy style to hltv
18:10:14 [INFO] Rollup damages set to False
18:10:14 [INFO] Parse frames set to True
18:10:14 [INFO] Running Golang parser from /home/ivan/.local/lib/python3.10/site-packages/csgo/parser/
18:10:14 [INFO] Looking for file at /home/ivan/.local/share/Steam/steamapps/common/Counter-Strike Global Offensive/csgo/testdem-2_2.dem
panic: runtime error: invalid memory address or nil pointer dereference
stacktrace:
goroutine 6 [running]:
runtime/debug.Stack()
	/usr/lib/go/src/runtime/debug/stack.go:24 +0x65
github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs.NewParserWithConfig.func1({0x7e8c20, 0xc002518500})
	/home/ivan/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v2@v2.9.1/pkg/demoinfocs/parser.go:321 +0x30
github.com/markus-wa/godispatch.(*Dispatcher).dispatchWithRecover.func1()
	/home/ivan/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:197 +0x42
panic({0x7e8c20, 0xc002518500})
	/usr/lib/go/src/runtime/panic.go:1038 +0x215
github.com/markus-wa/godispatch.callConsumerCode.func1()
	/home/ivan/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:122 +0x55
panic({0x7e8c20, 0xc0025184f0})
	/usr/lib/go/src/runtime/panic.go:1038 +0x215
github.com/markus-wa/godispatch.callConsumerCode.func1()
	/home/ivan/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:122 +0x55
panic({0x7c8d20, 0xc05200})
	/usr/lib/go/src/runtime/panic.go:1038 +0x215
github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs/common.(*TeamState).ClanName(...)
	/home/ivan/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v2@v2.9.1/pkg/demoinfocs/common/common.go:148
main.main.func25({0xc000001a00, 0xc000001860, 0x4a, 0x60, 0xc0000fa3c0, {0x0, 0x0}, 0x1a, 0x3, 0x1a, ...})
	/home/ivan/.local/lib/python3.10/site-packages/csgo/parser/parse_demo.go:2093 +0x93d
reflect.Value.call({0x7ad9c0, 0xc00011c440, 0xc0003b0d98}, {0x83d2d7, 0x4}, {0xc000ec3428, 0x1, 0x0})
	/usr/lib/go/src/reflect/value.go:543 +0x814
reflect.Value.Call({0x7ad9c0, 0xc00011c440, 0xc0000338c0}, {0xc000ec3428, 0x1, 0x1})
	/usr/lib/go/src/reflect/value.go:339 +0xc5
github.com/markus-wa/godispatch.callConsumerCode({0x7ad9c0, 0xc00011c440, 0x60}, {0xc000ec3428, 0x5000106, 0x100000000000000})
	/home/ivan/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:125 +0x4a
github.com/markus-wa/godispatch.(*Dispatcher).Dispatch(0xc000032300, {0x821a40, 0xc0000338c0})
	/home/ivan/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:111 +0x293
github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs.gameEventHandler.dispatch(...)
	/home/ivan/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v2@v2.9.1/pkg/demoinfocs/game_events.go:62
github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs.gameEventHandler.playerHurt({0xc0001369a0, 0x0, 0xc00010d140, 0xc00010d170}, 0xc0000c1000)
	/home/ivan/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v2@v2.9.1/pkg/demoinfocs/game_events.go:387 +0x645
github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs.(*parser).handleGameEvent(0xc0001369a0, 0xc00037a480)
	/home/ivan/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v2@v2.9.1/pkg/demoinfocs/game_events.go:41 +0xb3
reflect.Value.call({0x7ad080, 0xc00006c630, 0xc00010da40}, {0x83d2d7, 0x4}, {0xc000ec3f08, 0x1, 0x0})
	/usr/lib/go/src/reflect/value.go:543 +0x814
reflect.Value.Call({0x7ad080, 0xc00006c630, 0x200000080}, {0xc000ec3f08, 0x1, 0x1})
	/usr/lib/go/src/reflect/value.go:339 +0xc5
github.com/markus-wa/godispatch.callConsumerCode({0x7ad080, 0xc00006c630, 0x412a85}, {0xc000ec3f08, 0x412d11, 0xc000067478})
	/home/ivan/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:125 +0x4a
github.com/markus-wa/godispatch.(*Dispatcher).Dispatch(0xc0000322a0, {0x820cc0, 0xc00037a480})
	/home/ivan/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:111 +0x293
github.com/markus-wa/godispatch.(*Dispatcher).dispatchWithRecover(0x0, {0x820cc0, 0xc00037a480})
	/home/ivan/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:202 +0x5a
github.com/markus-wa/godispatch.(*Dispatcher).dispatchQueue(0xc0000322a0, 0x0)
	/home/ivan/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:177 +0xb1
created by github.com/markus-wa/godispatch.(*Dispatcher).AddQueues
	/home/ivan/go/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:162 +0x193


goroutine 1 [running]:
main.checkError(...)
	/home/ivan/.local/lib/python3.10/site-packages/csgo/parser/parse_demo.go:2345
main.main()
	/home/ivan/.local/lib/python3.10/site-packages/csgo/parser/parse_demo.go:2339 +0x2065
exit status 2
18:10:15 [INFO] Wrote demo parse output to testdem-2_2.json
18:10:15 [INFO] Reading in JSON from testdem-2_2.json
18:10:15 [INFO] JSON data loaded, available in the `json` attribute to parser
18:10:15 [INFO] Successfully parsed JSON output
18:10:15 [INFO] Successfully returned JSON output
```

**Desktop (please complete the following information):**
 - OS: Void linux
 - Version of python: Python 3.10 

**Additional context**
The `.json` file is still generated but it only contains the first few rounds

**Comment by pnxenopoulos** [comment]
Adding this here for posterity:

Looks like the linked demo is from a bunch of aim map rounds. The error is that `ClanName` must be nil. Easy fix.

**Comment by ivanhrabcak** [comment]
I am adding a second demo that is from a local match with bots, that is also failing to parse.

[Demo](https://www.mediafire.com/file/js33js31kug4n4v/testdemo.dem/file)

**Comment by pnxenopoulos** [comment]
Fixed in #75
